packages/cinterop/src/jvmCommon/kotlin/io/realm/interop/gc/NativeContext.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.interop.gc
+
+import io.realm.interop.LongPointerWrapper
+import java.lang.ref.ReferenceQueue
+
+object NativeContext {
+    private val referenceQueue: ReferenceQueue<LongPointerWrapper> = ReferenceQueue<LongPointerWrapper>()
+    private val finalizingThread = Thread(FinalizerRunnable(referenceQueue))
```

Comment:
I don't think we have a reliable `shutdown` event we could use to explicitly stop this Thread, since this cleaner is used across Threads, and we don't do the bookkeeping to detect that the Realm is closed all over the process's Threads to reliably close this cleaner.